src/ui/public/notify/notifier.js

```diff
@@ -449,23 +439,101 @@ Notifier.prototype.custom = function (msg, config, cb) {
     }
   };
 
-  const mergedConfig = _.assign({
+  const customConfig = _.assign({
     type: 'info',
     title: 'Notification',
-    content: formatMsg(msg, this.from),
     truncationLength: config.truncationLength || Notifier.config.defaultTruncationLength,
     lifetime: getLifetime(config.type)
   }, config);
 
-  const hasActions = _.get(mergedConfig, 'actions.length');
+  const hasActions = _.get(customConfig, 'actions.length');
   if (hasActions) {
-    mergedConfig.customActions = mergedConfig.actions;
-    delete mergedConfig.actions;
+    customConfig.customActions = customConfig.actions;
+    delete customConfig.actions;
   } else {
-    mergedConfig.actions = ['accept'];
+    customConfig.actions = ['accept'];
   }
 
-  return add(mergedConfig, cb);
+  return customConfig;
+}
+
+/**
+ * Display a custom message
+ * @param  {String} msg - required
+ * @param  {Object} config - required
+ * @param  {Function} cb - optional
+ *
+ * config = {
+ *   title: 'Some Title here',
+ *   type: 'info',
+ *   actions: [{
+ *     text: 'next',
+ *     callback: function() { next(); }
+ *   }, {
+ *     text: 'prev',
+ *     callback: function() { prev(); }
+ *   }]
+ * }
+ */
+Notifier.prototype.custom = function (msg, config, cb) {
+  const customConfig = customConfigHelper(config);
+  customConfig.content = formatMsg(msg, this.from);
+  return add(customConfig, cb);
+};
+
+/**
+ * Display a scope-bound directive using template rendering in the message area
+ * @param  {Object} directive - required
+ * @param  {Object} config - required
+ * @param  {Function} cb - optional
+ *
+ * directive = {
+ *  template: `<p>Hello World! <a ng-click="example.clickHandler()">Click me</a>.`,
+ *  controllerAs: 'example',
+ *  controller() {
+ *    this.clickHandler = () {
+ *      // do something
+ *    };
+ *  }
+ * }
+ *
+ * config = {
+ *   title: 'Some Title here',
+ *   type: 'info',
+ *   actions: [{
+ *     text: 'next',
+ *     callback: function() { next(); }
+ *   }, {
+ *     text: 'prev',
+ *     callback: function() { prev(); }
+ *   }]
+ * }
+ */
+Notifier.prototype.directive = function (directive, config, cb) {
```

Comment:
Are there any tests we can add for this? E.g. making sure the provided template gets rendered, that certain errors are thrown when bad arguments are provided.